game-core/src/main/java/games/strategy/triplea/ui/MapPanel.java

```diff
@@ -587,10 +586,9 @@ public void paint(final Graphics g) {
           final Optional<Image> image = uiContext.getUnitImageFactory().getHighlightImage(category.getType(),
               category.getOwner(), category.hasDamageOrBombingUnitDamage(), category.getDisabled());
           if (image.isPresent()) {
-            final AffineTransform t = new AffineTransform();
-            t.translate(normalizeX(r.getX() - getXOffset()) * scale, normalizeY(r.getY() - getYOffset()) * scale);
-            t.scale(scale, scale);
-            g2d.drawImage(image.get(), t, this);
+            final AffineTransform transform = AffineTransform.getScaleInstance(scale, scale);
```

Comment:
Does the ordering of translate and scale matter?

Reply:
Yes it does. 
That's why reordering the calls allowed to drop the scale param in the translate call.